src/routes/balances.rs

```diff
@@ -0,0 +1,18 @@
+use crate::utils::errors::ApiResult;
+use crate::utils::context::Context;
+use crate::config::request_cache_duration;
+use rocket::response::content;
+use crate::utils::cache::CacheExt;
+use crate::services::balances::*;
+use crate::models::service::balances::*;
+use crate::providers::info::DefaultInfoProvider;
+
+#[get("/v1/safes/<safe_address>/balances/<fiat>")]
+pub fn get_balances(context: Context, safe_address: String, fiat: String) -> ApiResult<content::Json<String>> {
+    let info_provider = DefaultInfoProvider::new(&context);
```

Comment:
In our transaction endpoint we build the info provider and so on in the `services` module. No strong opinion on that as for some endpoints without much logic we also do it in the route and it should not make any difference ... I think :P )

Reply:
It might be easier to test though 🤔  in this case we provide the exchange rate instead of having to use the mock...

Reply:
Not sure what's best

Reply:
If we go from the test perspective you could say more code is tested when you do it in the services (as we do not test the routes module, right?) :D

Reply:
well yeaaaah, but a counter argument could be made that mocked behaviour is synthetic and therefore having a more minimalist interface is more reliable. As in, passing the single float value we use instead of a full InfoProvider :)

Reply:
But then you still need to test how this single float value is calculated ... this is basically just pushing responsibilities. So I would agree with you if we test both methods, but we don't currently. Also the mocked behaviour is internally synthetic, not to the code calling it.

Reply:
Ah good point, If we'd eventually add tests for routes then yeah, we are just pushing that upwards. For the sake of consistency then I don't really mind pushing that down when I finally get to testing.